kpops/utils/gen_schema.py

```diff
@@ -96,9 +75,9 @@ def _add_components(
         components = tuple()
     # Set of existing types, against which to check the new ones
     defined_component_types: set[str] = {
-        component.__fields__["schema_type"].default
+        component.__fields__["type"].default
         for component in components
-        if component.__fields__.get("schema_type")
+        if component.__fields__.get("type")
```

Comment:
Would using the walrus operator be an improvement here?

```
defined_component_types: set[str] = {
    component_type.default
    for component in components
    if (component_type := component.__fields__.get("type"))
}

Reply:
absolutely, good point!

Reply:
actually it can just be
```python
defined_component_types = {
    component.get_component_type() for component in components
}
```
at this point every component has to have a type defined